ide/dlight.nativeexecution.nb/src/org/netbeans/modules/nativeexecution/ui/SSHKeyFileFilter.java

```diff
@@ -63,28 +60,28 @@ public boolean accept(File file) {
             return false;
         }
 
-        FileInputStream fis = null;
-        byte[] buffer = new byte[30];
-
         // fast sanity check
-        try {
-            fis = new FileInputStream(file);
-            fis.read(buffer, 0, 30);
 
-            Matcher m = p.matcher(new String(buffer, 0, 30, cs));
-            if (!m.matches()) {
+        // header contains the name of the tool so we don't know how long it is
+        // read until new line but with a limit and avoid buffering the key
+        int limit = 40;
+
+        try (InputStream is = new BufferedInputStream(Files.newInputStream(file.toPath()), limit)) {
+
+            StringBuilder head = new StringBuilder(limit);
+            int c;
```

Comment:
Instead of single-letter variables like c (character) and n (index), consider using currentChar or charIndex for improved readability, especially in longer functions. This makes the code's intent clearer without needing to trace variable definitions.
```suggestion
            int currentChar;
```

Reply:
I am ok with short variable names in cases like this here when the usage is only 1-5 lines away from the declaration. But I get what you are saying, I would have used a proper name if the scope of the variable would be wider.